cmd/pmm-managed-starlark/main_test.go

```diff
@@ -0,0 +1,157 @@
+	{"Value": "x86_64", "Variable_name": "version_compile_machine"},
+	{"Value": "Linux", "Variable_name": "version_compile_os"},
+	{"Value": "-log", "Variable_name": "version_suffix"},
+}
+
+func TestRunChecks(t *testing.T) {
+	testCases := []struct {
+		version      uint32
+		errorMessage string
+		stderr       string
+		stdout       string
+		name         string
+		script       string
+		scriptInput  []map[string]interface{}
+	}{
+
+		{
+			version:      1,
+			errorMessage: "exit status 1",
+			stderr:       invalidStarlarkScriptStderr,
+			stdout:       "",
+			name:         "invalid starlark script",
+			script:       "def check(): return []",
+			scriptInput:  validQueryActionResult,
+		},
+		{
+			version:      5,
+			errorMessage: "exit status 1",
+			stderr:       invalidVersionStderr,
+			stdout:       "",
+			name:         "invalid version",
+			script:       "def check(): return []",
+			scriptInput:  validQueryActionResult,
+		},
+		{
+			version:      1,
+			errorMessage: "exit status 2",
+			stderr:       memoryConsumingScriptStderr,
+			stdout:       "",
+			name:         "memory consuming starlark script",
+			script:       "def check(rows): return [1] * (1 << 30-1)",
+			scriptInput:  validQueryActionResult,
+		},
+		{
+			version:      1,
+			errorMessage: "signal: killed",
+			stderr:       "",
+			stdout:       "",
+			name:         "cpu consuming starlark script",
+			script: `def check(rows):
+						while True:
+							pass`,
+			scriptInput: validQueryActionResult,
+		},
+		{
+			version:      1,
+			errorMessage: "",
+			stderr:       "",
+			stdout:       "[{\"summary\":\"Fake check\",\"description\":\"Fake check description\",\"severity\":5,\"labels\":null}]\n",
+			name:         "valid starlark script",
+			script: `def check(rows):
+						results = []
+						results.append({
+							"summary": "Fake check",
+							"description": "Fake check description",
+							"severity": "warning",
+						})
+						return results`,
+			scriptInput: validQueryActionResult,
+		},
+	}
+
+	// since we run the binary as a child process to test it we need to build it first.
+	err := exec.Command("make", "-C", "../..", "release").Run()
+	require.NoError(t, err)
+
+	for _, tc := range testCases {
+		result, err := agentpb.MarshalActionQueryDocsResult(tc.scriptInput)
+		require.NoError(t, err)
+
+		data := checks.StarlarkScriptData{
+			CheckName:         tc.name,
+			CheckVersion:      tc.version,
+			Script:            tc.script,
+			QueryActionResult: result,
+		}
+
+		t.Run(tc.name, func(t *testing.T) {
+			cmd := exec.Command("./../../bin/pmm-managed-starlark")
+
+			var stdin, stderr bytes.Buffer
+			cmd.Stdin = &stdin
+			cmd.Stderr = &stderr
+			cmd.Env = []string{"PERCONA_TEST_STARLARK_ALLOW_RECURSION=true"}
+
+			encoder := json.NewEncoder(&stdin)
+			err = encoder.Encode(data)
+			require.NoError(t, err)
+
+			stdout, err := cmd.Output()
+			stderrContent := stderr.String()
+			if tc.stdout != "" {
```

Comment:
It looks like we don't need the branch – everything can be handled by assertions

Reply:
removed that branch, but I still need a branch to check for the errors and exit codes  because the last test case results in `nil` error